src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -141,13 +134,20 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
       const root : FiberRoot = (container.stateNode : any);
       const current = root.current;
 
-      root.pendingContext = getContextForSubtree(parentComponent);
+      const context = getContextForSubtree(parentComponent);
+      if (root.context === null) {
+        root.context = context;
+      } else {
+        root.pendingContext = context;
+      }
 
       scheduleTopLevelUpdate(current, element, callback);
 
       if (__DEV__) {
         if (ReactFiberInstrumentation.debugTool) {
-          if (element === null) {
+          if (current.type === null) {
```

Comment:
This is always true. If you wanted to check if this was a HostRoot, you'd check the tag, but even if you did this would always be true.

Instead you could probably check if there is an `.alternate` _before_ calling `scheduleTopLevelUpdate`. That's usually how we determine if something is initial mount or not.